src/functions/TelegramFunctions.cs

```diff
@@ -2,81 +2,87 @@
+    public class TelegramFunctions
     {
-      try
-      {
-        string body = await request.ReadAsStringAsync();
-        Update update = JsonConvert.DeserializeObject<Update>(body);
+        private readonly TelegramService telegramService;
+        private readonly ILogger<TelegramFunctions> logger;
 
-        if (!telegramService.IsValid(update))
+        public TelegramFunctions(
+            TelegramService telegramServiceParam,
+            ILogger<TelegramFunctions> loggerParam)
         {
-          return new ForbidResult();
+            this.telegramService = telegramServiceParam;
+            this.logger = loggerParam;
         }
 
-        (BuzzerEvent buzzerEvent, BuzzerEventPayload payload) = telegramService.GetBuzzerEventFromUpdate(update);
-
-        switch (buzzerEvent)
+        [FunctionName("TelegramHttpTrigger")]
+        public async Task<IActionResult> TelegramHttpTrigger(
+            [HttpTrigger(AuthorizationLevel.Anonymous, "post", Route = null)] HttpRequest request,
+            [Queue("acceptEntranceQueue")] QueueClient queueClient,
+            [Queue("accessLinkTokens")] QueueClient accessLinkTokenQueue,
+            [DurableClient] IDurableOrchestrationClient orchestrationClient)
         {
-          case BuzzerEvent.APPROVED:
-          case BuzzerEvent.REJECTED:
-            await BuzzerEventManager.RaiseBuzzerEvent(orchestrationClient, buzzerEvent, logger, payload);
-            break;
-          case BuzzerEvent.SCHEDULE_APPROVAL:
-            await QueueScheduledApproval(queueClient, update);
-            break;
-        }
-      }
-      catch (Exception e)
-      {
-        logger.LogError("[TelegramHttpTrigger] Exception: " + e.Message);
-        throw;
-      }
+            try
+            {
+                string body = await request.ReadAsStringAsync();
+                Update update = JsonConvert.DeserializeObject<Update>(body);
 
-      return new OkResult();
-    }
+                if (!telegramService.IsValid(update))
+                {
+                    return new ForbidResult();
+                }
 
-    private async Task QueueScheduledApproval(QueueClient queueClient, Update update)
-    {
-      int timeout;
-      var args = update.Message.Text.Split(' ');
+                (BuzzerEvent buzzerEvent, BuzzerEventPayload payload) = telegramService.GetBuzzerEventFromUpdate(update);
+
+                switch (buzzerEvent)
+                {
+                    case BuzzerEvent.APPROVED:
+                    case BuzzerEvent.REJECTED:
+                        await BuzzerEventManager.RaiseBuzzerEvent(orchestrationClient, buzzerEvent, logger, payload);
+                        break;
+                    case BuzzerEvent.SCHEDULE_APPROVAL:
+                        await QueueScheduledApproval(queueClient, update);
+                        break;
+                    case BuzzerEvent.GENERATE_ACCESS_LINK:
+                        Uri uri = new Uri(UriHelper.GetDisplayUrl(request));
```

Comment:
Maybe pass Uri/DisplayUrl/Request to GenerateAccessLink and encapsulate the rest of the logic there